demos/build_demos_msvc.bat

```diff
@@ -75,84 +77,54 @@ if "%PROCESSOR_ARCHITECTURE%" == "AMD64" (
-)
-
-if !VSWHERE! == "true" (
-    if "!VS_VERSION!"=="" (
-        echo Searching the latest Visual Studio...
-        for /f "usebackq tokens=*" %%i in (`vswhere -latest -products * -requires Microsoft.Component.MSBuild -property installationPath`) do (
-            set VS_PATH=%%i
-        )
+if "%VS_VERSION%" == "" (
+    if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" (
+        set "VSWHERE=%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+    ) else if exist "%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe" (
+        set "VSWHERE=%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
     ) else (
-        echo Searching Visual Studio !VS_VERSION!...
-        for /f "usebackq tokens=*" %%i in (`vswhere -products * -requires Microsoft.Component.MSBuild -property installationPath`) do (
-            set CUR_VS=%%i
-            if not "!CUR_VS:%VS_VERSION%=!"=="!CUR_VS!" (
-                set VS_PATH=!CUR_VS!
-            )
-        )
-    )
-    if exist "!VS_PATH!\MSBuild\14.0\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\14.0\Bin\MSBuild.exe"
-    )
-    if exist "!VS_PATH!\MSBuild\15.0\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\15.0\Bin\MSBuild.exe"
-    )
-    if exist "!VS_PATH!\MSBuild\Current\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\Current\Bin\MSBuild.exe"
+        echo vswhere.exe was not found.
+        echo Make sure Visual Studio 2017 version 15.2 or a newer version of Visual Studio
+        echo is installed.
+        goto errorHandling
     )
-)
 
-if "!MSBUILD_BIN!" == "" (
-    if "!VS_VERSION!"=="2015" (
-        if exist "C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" (
-            set "MSBUILD_BIN=C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe"
-            set "MSBUILD_VERSION=14 2015"
-        )
-    ) else if "!VS_VERSION!"=="2017" (
-        if exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe" (
-            set "MSBUILD_BIN=C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe"
-            set "MSBUILD_VERSION=15 2017"
-        ) else if exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe" (
-            set "MSBUILD_BIN=C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe"
-            set "MSBUILD_VERSION=15 2017"
-        ) else if exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe" (
-            set "MSBUILD_BIN=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe"
-            set "MSBUILD_VERSION=15 2017"
+    echo Searching for the latest Visual Studio...
+    for /f "usebackq tokens=*" %%i in (`"!VSWHERE!" -latest -products * -requires Microsoft.Component.MSBuild -property catalog_productLineVersion`) do (
```

Comment:
You have `-latest` key here. I expect that it reduces the output to only one version. The loop in not needed.

Reply:
The loop is needed, because that's the only way to get a program's output in the Windows shell.

Reply:
Okay